src/plugins/foxPage/components/MainOpportunity.tsx

```diff
@@ -36,6 +39,17 @@ export const MainOpportunity = ({
   })
   const hasActiveStaking = bnOrZero(foxyBalancesData?.opportunities?.[0]?.balance).gt(0)
 
+  const opportunityButtonTranslation = useMemo(() => {
+    if (hasActiveStaking) return 'plugins.foxPage.manage'
+    if (!wallet || !supportsETH(wallet)) return 'common.connectWallet'
```

Comment:
In theory we should never end in an undefined wallet state because of `isOpportunityButtonReady` but something something safety